### ISSUE TITLE
vmstorage: try optimize the performance for creating new time series.

### ISSUE THREAD

**Issue by waldoweng** [VictoriaMetrics/VictoriaMetrics#1244]
`vmstorage` slow down ingestion when a lots of new time series need to create.

After running some test on a 64-Core 256GB MEM machine, the maximum `churn rate` we get is around 150k/s per `vmstorage` instance, this is a huge performance degeneration compares to the ingestion rate when no new time series needs to be create, which is around 3mil datapoint/s per `vmstorage` instance.

I have done some profiling and found that, inside function `mergeset.Table.AddItem`,  a global mutex `mergeset.Table.rawItemsLock` is locked during the process of populating data to `mergeset.Table.rawItemsBlocks`, and this mutex limits the overall parallelism.

https://github.com/VictoriaMetrics/VictoriaMetrics/blob/97424c05c02aeb8fbb92205748be55b455c7710a/lib/mergeset/table.go#L354-L391

I have done some quick and dirty modification: https://github.com/VictoriaMetrics/VictoriaMetrics/pull/1243/commits/5f614af0727837bca862f566742f9dd52e52e83d, split `mergeset.Table.rawItemsBlocks` to 64 parts and try eliminate the use of `mergeset.Table.rawItemsLock`, and the result seems amazing.

benchmark on my mac shows roughly 3 times performance up, and benchmark on our test machine shows 6-7 times performance up

* benchmark on mac with branch origin/cluster
```bash
go test -benchmem -run=^$ -bench "^(BenchmarkIndexDBAddTSIDs)$" github.com/VictoriaMetrics/VictoriaMetrics/lib/storage -v
goos: darwin
goarch: amd64
pkg: github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
BenchmarkIndexDBAddTSIDs
2021-04-24T08:35:44.843Z	info	lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:35:44.869Z	info	lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.027 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:35:44.869Z	info	lib/memory/memory.go:37	limiting caches to 10307921510 bytes, leaving 6871947674 bytes to the OS according to -memory.allowedPercent=60
2021-04-24T08:35:44.872Z	info	lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:44.873Z	info	lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:44.873Z	info	lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:44.880Z	info	lib/mergeset/table.go:227	converters stopped in 0.007 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:44.880Z	info	lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:44.880Z	info	lib/mergeset/table.go:232	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:44.880Z	info	lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:35:44.954Z	info	lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.074 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:44.960Z	info	lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:35:44.975Z	info	lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.014 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:227	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:232	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:45.072Z	info	lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:35:45.784Z	info	lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.713 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:45.803Z	info	lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:35:45.818Z	info	lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.015 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:35:48.171Z	info	lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:48.171Z	info	lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:48.171Z	info	lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:48.171Z	info	lib/mergeset/table.go:227	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:48.171Z	info	lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:35:48.207Z	info	lib/mergeset/table.go:232	part mergers stopped in 0.036 seconds on "bench-index-db-add-tsids"
2021-04-24T08:35:48.207Z	info	lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:35:48.639Z	info	lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.432 seconds on "bench-index-db-add-tsids"
BenchmarkIndexDBAddTSIDs-12    	    1245	   1889400 ns/op	   0.53 MB/s	  507067 B/op	     148 allocs/op
PASS
ok  	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage	4.997s
```

* benchmark on mac with modification
```
go test -benchmem -run=^$ -bench "^(BenchmarkIndexDBAddTSIDs)$" github.com/VictoriaMetrics/VictoriaMetrics/lib/storage -v
goos: darwin
goarch: amd64
pkg: github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
BenchmarkIndexDBAddTSIDs
2021-04-24T08:34:24.215Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:34:24.234Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.019 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:34:24.234Z	info	lib/memory/memory.go:37	limiting caches to 10307921510 bytes, leaving 6871947674 bytes to the OS according to -memory.allowedPercent=60
2021-04-24T08:34:24.240Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.240Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.240Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.244Z	info	lib/mergeset/table.go:249	converters stopped in 0.005 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.244Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.244Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.244Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.245Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.252Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:34:24.266Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.014 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:249	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:34:24.332Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:24.337Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:34:24.349Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.011 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:249	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:34:25.368Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
BenchmarkIndexDBAddTSIDs-12    	    1852	    549784 ns/op	   1.82 MB/s	  757062 B/op	    2120 allocs/op
PASS
ok  	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage	2.362s
```

* benchmark on test machine with branch origin/cluster
```bash
go test -benchmem -run=^$ -bench "^(BenchmarkIndexDBAddTSIDs)$" github.com/VictoriaMetrics/VictoriaMetrics/lib/storage -v
goos: linux
goarch: amd64
pkg: github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
BenchmarkIndexDBAddTSIDs
2021-04-24T08:55:05.277Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:55:05.278Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:55:05.283Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/memory/memory.go:37	limiting caches to 161764796006 bytes, leaving 107843197338 bytes to the OS according to -memory.allowedPercent=60
2021-04-24T08:55:05.288Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.288Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.289Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.289Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:227	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.289Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.289Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:232	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.289Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.345Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.057 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.355Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:55:05.356Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:227	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:232	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:05.574Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:55:06.571Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.997 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:06.584Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:169	opening table "bench-index-db-add-tsids"...
2021-04-24T08:55:06.584Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:203	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:55:09.131Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:219	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:09.131Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:222	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:09.131Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:224	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:09.131Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:227	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:09.131Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:229	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:55:09.168Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:232	part mergers stopped in 0.036 seconds on "bench-index-db-add-tsids"
2021-04-24T08:55:09.168Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:234	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:55:09.225Z	info	/home/toc/SDE/VictoriaMetrics_ori/lib/mergeset/table.go:258	0 inmemory parts have been flushed to files in 0.057 seconds on "bench-index-db-add-tsids"
BenchmarkIndexDBAddTSIDs-64    	     553	   4602753 ns/op	   0.22 MB/s	  565702 B/op	     179 allocs/op
PASS
ok  	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage	4.256s
```

* benchmark on test machine with modification

```
go test -benchmem -run=^$ -bench "^(BenchmarkIndexDBAddTSIDs)$" github.com/VictoriaMetrics/VictoriaMetrics/lib/storage -v
goos: linux
goarch: amd64
pkg: github.com/VictoriaMetrics/VictoriaMetrics/lib/storage
BenchmarkIndexDBAddTSIDs
2021-04-24T08:56:01.901Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:56:01.902Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:56:01.908Z	info	lib/memory/memory.go:37	limiting caches to 161764796006 bytes, leaving 107843197338 bytes to the OS according to -memory.allowedPercent=60
2021-04-24T08:56:01.917Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.917Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.917Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.918Z	info	lib/mergeset/table.go:249	converters stopped in 0.002 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.918Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.918Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.918Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.918Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.924Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:56:01.925Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:249	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:56:01.991Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:02.003Z	info	lib/mergeset/table.go:175	opening table "bench-index-db-add-tsids"...
2021-04-24T08:56:02.004Z	info	lib/mergeset/table.go:209	table "bench-index-db-add-tsids" has been opened in 0.001 seconds; partsCount: 0; blocksCount: 0, itemsCount: 0; sizeBytes: 0
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:241	waiting for raw items flusher to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:244	raw items flusher stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:246	waiting for converters to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:249	converters stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:251	waiting for part mergers to stop on "bench-index-db-add-tsids"...
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:254	part mergers stopped in 0.000 seconds on "bench-index-db-add-tsids"
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:256	flushing inmemory parts to files on "bench-index-db-add-tsids"...
2021-04-24T08:56:03.033Z	info	lib/mergeset/table.go:280	0 inmemory parts have been flushed to files in 0.000 seconds on "bench-index-db-add-tsids"
BenchmarkIndexDBAddTSIDs-64    	    1882	    545499 ns/op	   1.83 MB/s	  777525 B/op	    2121 allocs/op
PASS
ok  	github.com/VictoriaMetrics/VictoriaMetrics/lib/storage	1.478s
```

please take a look. thx

**Comment by valyala** [comment]
@waldoweng , thanks for the idea and initial implementation! Could you verify whether the following commits improve the registration for new time series on a system with many CPU cores in the same way as the proposed implementation at https://github.com/VictoriaMetrics/VictoriaMetrics/pull/1243 ?

* For single-node VictoriaMetrics - 56b6b893ce821eaa14ce6a6730b26e0f342b3670
* For cluster version of VictoriaMetrics - 2d1d60118d0d49e2bc98f19ed4b1acecd2048975

**Comment by waldoweng** [comment]
I build and run the benchmark, the performance is amazing, and memory allocation is far more less than the prototype. looking forward to the release!

**Comment by davidsome** [comment]
amazing !!!

**Comment by valyala** [comment]
Thanks for testing the commit and providing the feedback on it! The commit will be included in the next release of VictoriaMetrics.

**Comment by valyala** [comment]
The optimization has been included in [v1.59.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as done.
